tests/func-tests/observability_controller_test.go

```diff
@@ -55,11 +86,53 @@ var _ = Describe("Observability Controller", Label(tests.OpenshiftLabel, testNam
 
 			// Wait for the controller to recreate the silence
 			Eventually(func() bool {
-				amSilences, err := amApi.ListSilences()
+				amSilences, err := amAPI.ListSilences()
 				Expect(err).ToNot(HaveOccurred())
 
 				return observability.FindPodDisruptionBudgetAtLimitSilence(amSilences) != nil
 			}, "5m", "10s").Should(BeTrue())
 		})
 	})
 })
+
+func isAlertmanagerAccessible(httpClient http.Client, svcHost string, bearerToken string) bool {
+	req, err := http.NewRequest("GET", svcHost+"/api/v1/status", nil)
+	if err != nil {
+		return false
+	}
+
+	if bearerToken != "" {
+		req.Header.Add("Authorization", "Bearer "+bearerToken)
+	}
+
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		return false
+	}
+	defer resp.Body.Close()
+
+	return resp.StatusCode < 500
```

Comment:
if the request returns HTTP status code of 4xx, why it is considered that the alert manager is accessible?
probably the endpoint is not available, or you're not authorized or other reasons.
I think we want the response to be at 2xx, no?

Reply:
@orenc1 updated